src/hotspot/cpu/x86/x86.ad

```diff
@@ -7831,6 +7831,22 @@ instruct vblendvpFD(legVec dst, legVec src1, legVec src2, legVec mask) %{
   ins_pipe( pipe_slow );
 %}
 
+instruct ablendvp(legVec dst, legVec src1, legVec src2, legVec mask, legVec vtmp) %{
+  predicate(UseAVX > 0 && EnableX86ECoreOpts &&
+            n->in(2)->bottom_type()->isa_vectmask() == NULL &&
+            Matcher::vector_length_in_bytes(n) <= 32);
+  match(Set dst (VectorBlend (Binary src1 src2) mask));
+  format %{ "vector_blend  $dst,$src1,$src2,$mask\t! using $vtmp as TEMP" %}
+  effect(TEMP vtmp, TEMP dst);
```

Comment:
TEMP dst can be removed.

Reply:
TEMP dst is needed because otherwise register allocator can allocate dst and vtmp to be same. The vpand will then overwrite vtmp.